core/dbt/context/providers.py

```diff
@@ -1096,38 +1148,23 @@ def adapter_macro(self, name: str, *args, **kwargs):
                ...
             {%- endmacro %}
         """
-        original_name: str = name
-        package_name: Optional[str] = None
+        deprecations.warn('adapter-macro', macro_name=name)
+        package_names: Optional[List[str]] = None
         if '.' in name:
             package_name, name = name.split('.', 1)
+            package_names = [package_name]
 
-        attempts = []
-
-        for prefix in self._get_adapter_macro_prefixes():
-            search_name = f'{prefix}__{name}'
-            try:
-                macro = self.namespace.get_from_package(
-                    package_name, search_name
-                )
-            except CompilationException as exc:
-                raise CompilationException(
-                    f'In adapter_macro: {exc.msg}, original name '
-                    f"'{original_name}'",
-                    node=self.model,
-                ) from exc
-            if package_name is None:
-                attempts.append(search_name)
-            else:
-                attempts.append(f'{package_name}.{search_name}')
-            if macro is not None:
-                return macro(*args, **kwargs)
-
-        searched = ', '.join(repr(a) for a in attempts)
-        raise_compiler_error(
-            f"In adapter_macro: No macro named '{name}' found\n"
-            f"    Original name: '{original_name}'\n"
-            f"    Searched for: {searched}"
-        )
+        try:
+            macro = self.db_wrapper.dispatch(
+                macro_name=name, packages=package_names
+            )
+        except CompilationException as exc:
+            raise CompilationException(
+                f'In adapter_macro: {exc.msg}\n'
+                f"    Original name: '{name}'",
```

Comment:
Could we make this the full original macro name, including the package namespace if it included a `.`?